src/include/sof/audio/component.h

```diff
@@ -702,11 +702,20 @@ int comp_set_state(struct comp_dev *dev, int cmd);
 static inline void component_set_period_frames(struct comp_dev *current,
 					       uint32_t rate)
 {
-	/* Samplerate is in Hz and period in microseconds.
+	/* Sample rate is in Hz and period in microseconds.
 	 * As we don't have floats use scale divider 1000000.
 	 * Also integer round up the result.
 	 */
-	current->frames = ceil_divide(rate * current->period, 1000000);
+	if (rate % 44100 == 0) {
```

Comment:
faster to use a switch statement here as % is expensive and rate will be 44100 or 88200, also avoids the divide too

Reply:
sure , updated